Automattic/jetpack
# Add caching for VideoPress functions with meta queries

#### Changes proposed in this Pull Request:

* Caches functions that use `WP_Query`
* Adds some arguments to `WP_Query` to improve performance, like `'no_found_rows'  => true`
* Adds a few unit tests

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* Enhances an existing feature, no intended change to functionality

#### Testing instructions:
(Regression testing for VideoPress)
1. Ensure VideoPress is enabled:
![settings-jetpack](https://user-images.githubusercontent.com/4063887/75300287-04608c00-57fd-11ea-8134-362c1546c824.png)

2. Create a new post
3. Add a Video block
4. Upload a video to it
5. Expected: the video plays in the editor.
6. Expected: the video plays on the front-end

#### Proposed changelog entry for your changes:

* Performance enhancement for VideoPress, caching and improving queries
